File: python/1544.make_the_string_great.py
class Solution:
    def makeGood(self, s: str) -> str:

        my_stack = []

        for element in s:
            if my_stack and abs(ord(element) - ord(my_stack[-1])) == 32:
                my_stack.pop()
            else:
                my_stack.append(element)

        return "".join(my_stack)

File: cpp/1544.make_the_string_great.cpp
#include <string>
#include <vector>

class Solution {
 public:
    std::string makeGood(std::string s) {
        std::vector<char> list;

        for (char ch : s) {
            if (!list.empty() && std::abs(static_cast<int>(list.back()) -
                                          static_cast<int>(ch)) == 32) {
                list.pop_back();
            } else {
                list.push_back(ch);
            }
        }
        std::string result;
        for (char ch : list) {
            result.push_back(ch);
        }
        return result;
    }
};
